src/pages/address/components/TokenSearchDropDown.tsx

```diff
@@ -0,0 +1,129 @@
+      </div>
+    </div>
+  );
+}
+
+export default function TokenSearchDropDown({
+  addressTokens,
+}: TokenDropDownProps) {
+  const [isFocused, setIsFocused] = useState<boolean>(false);
+  const [isSearching, setIsSearching] = useState<boolean>(false);
+  const [searchString, setSearchString] = useState("");
+  const [searchedList, setSearchedList] = useState(addressTokens);
+
+  useEffect(() => {
+    const displayTokens: AddressToken[] = [];
+
+    const getSearchedString = async () => {
+      setIsSearching(true);
+      await sleep(300);
+      for (let i = 0; i < addressTokens.length; i += 1) {
+        if (addressTokens[i].symbol.toLowerCase().includes(searchString)) {
+          displayTokens.push(addressTokens[i]);
+        }
+      }
+      setIsSearching(false);
+    };
+
+    if (searchString !== "") {
+      getSearchedString();
+      setSearchedList(displayTokens);
+    } else {
+      setSearchedList(addressTokens);
+    }
+  }, [searchString]);
+
+  return (
+    <GradientCardContainer className="absolute whitespace-nowrap translate-x-[-10px] translate-y-[13px] lg:w-[314px] lg:h-[284px] md:w-[314px] md:h-[284px] w-[296px] h-[300px]">
+      <div
+        className={clsx(
+          "p-5 w-full h-full overflow-y-hidden hover:overflow-y-auto",
+          { "hover:overflow-y-hidden": searchedList.length <= 3 }
+        )}
+        onBlur={() => {
+          setIsFocused(false);
+        }}
+        onFocus={() => {
+          setIsFocused(true);
+        }}
+      >
+        <div
+          className={clsx(
+            "relative group flex w-full rounded-lg p-[0.5px] mb-5 bg-black-500 focus-within:bg-lightBlue black-gradient-1-shadow backdrop-blur-[6px]",
+            "transition duration-300 ease-in"
+          )}
+        >
+          {!isFocused && (
+            <div
+              className={clsx(
+                "absolute opacity-0 inset-0 rounded-lg transition brand-gradient-1 group-hover:opacity-100 z-[-10]",
+                "transition duration-300 ease-in"
+              )}
+            />
+          )}
+          <div className="flex flex-row items-center rounded-lg bg-black-900 black-gradient-1 gap-x-[13px] pt-[17px] pb-[19px]">
+            <FiSearch className="ml-[23px] text-white-50" size={24} />
+            <input
+              className="w-2/3 h-full focus:outline-none bg-black-900 black-gradient-1 border-none black-gradient-1-shadow text-white-700 text-xl focus:caret-lightBlue"
+              onChange={(v) => setSearchString(v.target.value)}
+              placeholder="Search..."
+              value={searchString}
+            />
+            <IoCloseCircleSharp
+              onClick={() => {
+                setSearchString("");
+              }}
+              role="button"
+              className={clsx(
+                "text-white-50 mr-[18px] opacity-0",
+                {
+                  "opacity-100": isFocused,
+                },
+                { "opacity-100": searchString !== "" }
```

Comment:
can we combine the conditions?

Reply:
actually can we not use `opacity`, we conditionally display the entire element instead

Reply:
updated to conditionally display the entire element